cnxpublishing/views/admin.py

```diff
@@ -522,3 +526,169 @@ def admin_content_status_single_POST(request):
+                '{} is not a valid version'.format(version))
+        if len(version.split(".")) > 1:
+            try:
+                minor = int(version.split(".")[1])
+            except ValueError:
+                raise httpexceptions.HTTPBadRequest(
+                    '{} is not a valid version'.format(version))
+
+    with db_connect() as db_conn:
+        with db_conn.cursor() as cursor:
+
+            module_ident = ""
+            if (major and minor):
+                cursor.execute("""\
+                    SELECT module_ident from modules as m
+                    WHERE m.uuid =%s and m.major_version=%s and
+                        m.minor_version=%s
+                    ;""", vars=(uuid, major, minor))
+            elif(major and not minor):
+                cursor.execute("""\
+                    SELECT module_ident, minor_version from modules as m
+                    WHERE m.uuid =%s and m.major_version=%s
+                    ORDER BY revised desc LIMIT 1;
+                    ;""", vars=(uuid, major))
+            else:  # just uuid
+                cursor.execute("""\
+                    SELECT module_ident, major_version, minor_version
+                    from modules as m WHERE m.uuid =%s
+                    ORDER BY revised desc LIMIT 1;
+                    ;""", vars=(uuid, ))
+            modules_found = cursor.fetchall()
+            try:
+                module_ident = modules_found[0][0]
+            except IndexError:
+                raise httpexceptions.HTTPBadRequest(
+                    '{} is not an existing module'.format(ident_hash))
+            try:
+                major = module_ident = modules_found[0][1]
+                minor = module_ident = modules_found[0][2]
+            except IndexError:
+                pass
+            filename = request.POST['picture'].filename
+            input_file = request.POST['picture'].file
+            input_file.seek(0)
+            f = input_file.read()
+            file_bytes = bytearray(f)
+            file_sha1 = hashlib.new('sha1', file_bytes).hexdigest()
+
+            cursor.execute("""SELECT sha1 from files WHERE sha1 = %s;""",
+                           vars=(file_sha1, ))
+            files_found = len(cursor.fetchall())
+            response = ""
+            if files_found > 0:
+                # the file is already in the database
+                response += "Image file already in the databse. "
+            else:
+                # have to add the file
+                cursor.execute("""\
+                    INSERT INTO files (file, media_type)
+                    VALUES (%s, 'image/png')
+                    RETURNING fileid;
+                    """, vars=(file_bytes, ))
+                fileid = cursor.fetchall()[0][0]
+                cursor.execute("""\
+                    INSERT INTO module_files (module_ident, fileid, filename)
+                    VALUES (%s, %s, 'featured-cover.png')
```

Comment:
Instead of using magic filename, let's just keep the fileid in the featured_books table.